pkg/operator/ceph/file/controller.go

```diff
@@ -255,6 +255,11 @@ func (r *ReconcileCephFilesystem) reconcileCreateFilesystem(cephFilesystem *ceph
 		return reconcile.Result{}, errors.Wrapf(err, "failed to get controller %q owner reference", cephFilesystem.Name)
 	}
 
+	// preservePoolsOnDelete being set to true does not make sense because of data-loss concerns (see #6492).
+	if cephFilesystem.Spec.PreservePoolsOnDelete {
+		return reconcile.Result{}, errors.Wrapf(err, "preservePoolsOnDelete has been deprecated. Set preserveFilesystemOnDelete instead")
```

Comment:
```suggestion
		return reconcile.Result{}, errors.New("preservePoolsOnDelete has been deprecated. Set preserveFilesystemOnDelete instead")
```

Reply:
This is no `err` at this point to return, it's a brand new error.

Reply:
Done